chemie/corporate/views.py

```diff
@@ -130,3 +168,197 @@ def job_remove(request, id):
+        question.delete()
+        return redirect("corporate:question_create")
+
+    else:
+        return redirect("corporate:statistics")
+
+
+@permission_required("corporate:add_surveyquestion")
+def add_question_to_survey(request, year):
+    survey = get_object_or_404(Survey, year=year)
+
+    if request.method == "POST":
+        for key, value in request.POST.items():
+            if "checkbox" in key:
+                question = get_object_or_404(
+                    SurveyQuestion, question=request.POST[key]
+                )
+                # The way the database is organized now, a survey can't be
+                # linked to a surveyquestion without there being any
+                # answers. This creates a placeholder answer so it
+                # shows up on the survey admin page
+                default_answer = AnswerKeyValuePair.objects.create(
+                    key="Placeholder for svaralternativ",
+                    value=0,
+                    question=question,
+                    survey=survey,
+                )
+                default_answer.save()
+        return redirect("corporate:survey_edit", year)
+
+    survey_questions = survey.get_q_a_dict().keys()
+    all_questions = SurveyQuestion.objects.values_list("question", flat=True)
+
+    questions = []
+    for question in all_questions:
+        if question not in survey_questions:
+            questions.append(question)
+
+    context = {"survey": survey, "questions": questions}
+
+    return render(request, "corporate/statistics_add_question.html", context)
+
+
+@permission_required("corporate:add_answerkeyvaluepair")
+def answer_create(request, year, question):
+    survey = get_object_or_404(Survey, year=year)
+    question = get_object_or_404(SurveyQuestion, question=question)
+    form = CreateAnswerForm(request.POST or None)
+
+    # Survey and question fields are already determined and can't be changed
+    form.fields.pop("survey")
+    form.fields.pop("question")
+
+    if request.method == "POST":
+        if form.is_valid():
+            data = form.cleaned_data
+            new_answer = AnswerKeyValuePair(
+                key=data["key"],
+                value=data["value"],
+                question=question,
+                survey=survey,
+            )
+            new_answer.save()
+            return redirect("corporate:survey_edit", year=survey.year)
+
+    context = {"survey": survey, "question": question, "form": form}
+    return render(request, "corporate/statistics_answer_create.html", context)
+
+
+@permission_required("corporate:change_answerkeyvaluepair")
+def answer_edit(request):
+    if request.method == "POST":
+        id = request.POST["id"]
+        value = request.POST["value"]
+        answer = request.POST["answer"]
+
+        answer_value_pair = get_object_or_404(AnswerKeyValuePair, id=id)
+        answer_value_pair.value = value
+        answer_value_pair.answer = answer
+
```

Comment:
Change value and answer to edited_value and edited_answer, respectively (I think)